src/main/java/io/debezium/connector/vitess/VitessConnectorConfig.java

```diff
@@ -216,6 +232,28 @@ public Duration getKeepaliveInterval() {
         return getConfig().getDuration(KEEPALIVE_INTERVAL_MS, ChronoUnit.MILLIS);
     }
 
+    public Map<String, String> getGrpcHeaders() {
```

Comment:
This seems less ideal since we would construct a new map every time we call this. An alternative could be having a private `Map<String, String> grpcHeader` in VitessConnectorConfig and populate like what we do here in the constructor. Then the `getGrpcHeaders` will return that private map directly.

Given this is only being called in one place and we know it will be called once only - I don't feel strong either way, would leave it to you

Reply:
Yes. It's called only once during VStream start. I saw couple of places in other *Config class do this (construct a new map in the method) so I follow the same pattern.

Reply:
Yeah, shouldn't matter much in this case. In general, we tend to extract and build up that config state once in the constructor in cases where this is a more involved operation and simply pass-through to the underlying config if it's a simple read. In that light, I'd probably have extracted that map creation and stored into a field of this class, but as you say it only is called once, so there's no tangible impact at this point.